test/e2e/common/setup-kubestellar.sh

```diff
@@ -86,27 +86,29 @@ echo "wds1 created."
 
 :
 : -------------------------------------------------------------------------
-: Run OCM transport controller executable
+: Run OCM transport controller in a pod
 :
 cd "${SRC_DIR}/../../.." ## go up to KubeStellar directory
 KUBESTELLAR_DIR="$(pwd)"
-## this is a temp solution - run as executble process. this should be replaced to run as pod, ideally using helm
-OCM_TRANSPORT_PLUGIN_RELEASE="0.1.0-rc2"
+OCM_TRANSPORT_PLUGIN_RELEASE="0.1.0-rc4"
 curl -sL https://github.com/kubestellar/ocm-transport-plugin/archive/refs/tags/v${OCM_TRANSPORT_PLUGIN_RELEASE}.tar.gz | tar xz
 cd ocm-transport-plugin-${OCM_TRANSPORT_PLUGIN_RELEASE}
 OCM_TRANSPORT_PLUGIN_DIR="$(pwd)"
 pwd
 echo "replace github.com/kubestellar/kubestellar => ${KUBESTELLAR_DIR}/" >> go.mod
-go mod tidy
-make build
-mv ./bin/ocm-transport-plugin ${KUBESTELLAR_DIR}/ocm-transport-plugin
+IMAGE_TAG=${OCM_TRANSPORT_PLUGIN_RELEASE} make ko-build-local
+kind load --name kubeflex docker-image ko.local/transport-controller:${OCM_TRANSPORT_PLUGIN_RELEASE} # load local image to kubeflex
 cd "${KUBESTELLAR_DIR}"
 pwd
 rm -rf ${OCM_TRANSPORT_PLUGIN_DIR}
 echo "running ocm transport plugin..."
-./ocm-transport-plugin --transport-context imbs1 --wds-context wds1 --wds-name wds1 -v=4 &> transport.log &
+kubectl config use-context kind-kubeflex ## transport deployment script assumes it runs within kubeflex context
+IMAGE_PULL_POLICY=Never ./hack/deploy-transport-controller.sh wds1 imbs1 ko.local/transport-controller:${OCM_TRANSPORT_PLUGIN_RELEASE}
 
-echo "transport controller is running as background process."
+kubectl -n wds1-system wait --for=condition=Ready pod/$(kubectl -n wds1-system get pods -l name=transport-controller -o jsonpath='{.items[*].metadata.name}')
```

Comment:
The `*` here is dangerous.

`kubectl wait` will immediately fail if the relevant object does not exist yet. That is why the bash scripts have their own wait loop that allows the object to not exist for a while.

Reply:
fixed

Reply:
I see that one of the two things I raised got fixed. I still see the `*`.